pkg/kubectl/scale.go

```diff
@@ -177,7 +177,11 @@ func (scaler *ReplicationControllerScaler) Scale(namespace, name string, newSize
 		return err
 	}
 	if waitForReplicas != nil {
-		watchOptions := api.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", name), ResourceVersion: "0"}
+		latestRC, err := scaler.c.ReplicationControllers(namespace).Get(name)
```

Comment:
the way this function is written, it just guarantees that rc becomes stable, not that it stabilizes for this scale update. 
This get might be after the rc manager updates status with what you want. To actually see your update result you need to watch from the rv of your update (if you updated rv from 10->11, watch at 11). This still doesn't guarantee that you see *your* update result, because the rc manager could batch updates (i.e rv 11 is your update of replicas from 10 to 11, rv 12 is another update of replicas to 2000, rc manager can squash 11 and 2000 and start scaling to 2000 which you end up waiting for)